### PR TITLE
feat: Added dev and prod targets.

### PR DESCRIPTION
**Pull request by zubairshakoorarbisoft**
Issue: https://github.com/openedx/edx-cookiecutters/issues/292
**Merge checklist:**
Check off if complete *or* not applicable:
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Fixup commits are squashed away
- [ ] Unit tests added/updated
- [ ] Manual testing instructions provided
- [ ] Noted any: Concerns, dependencies, deadlines, tickets

### CODE REVIEW

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile [hunk 1]
```diff
@@ -39,13 +39,16 @@ RUN locale-gen en_US.UTF-8
...
 
 WORKDIR /edx/app/{{cookiecutter.repo_name}}
 
+FROM app as prod
+
+ENV DJANGO_SETTINGS_MODULE {{cookiecutter.project_name}}.settings.production
+
 # Copy the requirements explicitly even though we copy everything below
 # this prevents the image cache from busting unless the dependencies have changed.
 COPY requirements/production.txt /edx/app/{{cookiecutter.repo_name}}/requirements/production.txt
```
Comment (timmc-edx) [inline cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile:54]: Does this instruction need to be duplicated for the dev target as well? Or is dev.txt getting copied somewhere else?

Comment (zubairshakoorarbisoft) [reply cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile:54]: Noted, added steps to copy `dev.txt` and the `code`

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile [hunk 2]
```diff
@@ -181,18 +181,14 @@ db-shell: # Run the app shell as root, enter the app's database
...
-
-github_docker_auth:
+docker_auth:
 	echo "$$DOCKERHUB_PASSWORD" | docker login -u "$$DOCKERHUB_USERNAME" --password-stdin
 
-github_docker_push: github_docker_tag github_docker_auth ## push to docker hub
-	docker push 'openedx/{{cookiecutter.repo_name}}:latest'
-	docker push "openedx/{{cookiecutter.repo_name}}:${GITHUB_SHA}"
+docker_tag_build_push: docker_auth
+	docker buildx build -t openedx/{{cookiecutter.repo_name}}:latest -t openedx/{{cookiecutter.repo_name}}:${GITHUB_SHA} --platform linux/amd64,linux/arm64 --build-arg SERVICE_VARIANT={{cookiecutter.repo_name}} --build-arg SERVICE_PORT={{cookiecutter.port}} --target production --push .
```
Comment (timmc-edx) [inline cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile:188]: In these recipes, what are the role of SERVICE_VARIANT and SERVICE_PORT? It looks like they're unused. I think only edxapp uses service variants (lms vs cms) and port should be set using cookiecutter.

**Review by timmc-edx — APPROVED** [review]

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile [hunk 3]
```diff
@@ -65,3 +68,21 @@ CMD gunicorn --workers=2 --name {{cookiecutter.repo_name}} -c /edx/app/{{cookiec
...
+
+RUN pip install -r requirements/dev.txt
+
+# After the requirements so changes to the code will not bust the image cache
+COPY . /edx/app/{{cookiecutter.repo_name}}
+
+# Devstack related step for backwards compatibility
+RUN touch {{cookiecutter.project_name}}/{{cookiecutter.project_name}}_env
+
+CMD while true; do python ./manage.py runserver 0.0.0.0:{{cookiecutter.port}}; sleep 2; done
```
Comment (iamsobanjaved) [inline cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile:88]: Nit: add blank line at the end of file

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile [hunk 4]
```diff
@@ -181,18 +181,14 @@ db-shell: # Run the app shell as root, enter the app's database
...
 	echo "$$DOCKERHUB_PASSWORD" | docker login -u "$$DOCKERHUB_USERNAME" --password-stdin
 
-github_docker_push: github_docker_tag github_docker_auth ## push to docker hub
-	docker push 'openedx/{{cookiecutter.repo_name}}:latest'
-	docker push "openedx/{{cookiecutter.repo_name}}:${GITHUB_SHA}"
+docker_tag_build_push: docker_auth
+	docker buildx build -t openedx/{{cookiecutter.repo_name}}:latest -t openedx/{{cookiecutter.repo_name}}:${GITHUB_SHA} --platform linux/amd64,linux/arm64 --target production --push .
+
+docker_tag_build_push_dev: docker_auth
+	docker buildx build -t openedx/{{cookiecutter.repo_name}}-dev:latest -t openedx/{{cookiecutter.repo_name}}-dev:${GITHUB_SHA} --platform linux/amd64,linux/arm64 --target development --push .
```
Comment (iamsobanjaved) [inline cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile:191]: You have defined the target in Dockerfile as `dev` so this should be `dev` as well.

**Review by iamsobanjaved — APPROVED** [review]
Just two changes are needed, otherwise looks fine.

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile (hunk 4 above)
Comment (timmc-edx) [reply cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile:191]: Ooh, that's a good catch! Same for prod/production.

Comment (zubairshakoorarbisoft) [reply cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile]: Thanks @iamsobanjaved. It's fixed.

**Pull request merged**